### PR TITLE
Add and generate tin ore

### LINKED ISSUE #3: Generate tin ore
**Describe the solution you'd like**
Generate tin ore.

**Additional context**
Can refer TR.

### PR DESCRIPTION
**Pull request by longfangsong**
Every industry mod has tin ore, we can add this too.

Currently:
- World generator can generate tin ore in 0~64 layer overworld.
- We can dig the ore block up and get raw tin item.

Close #3.

Reference:
https://fabricmc.net/wiki/tutorial:ores

图片
图片

### CODE REVIEW

File: src/main/kotlin/com/fairchild_superconductor/magic_crystal/MagicCrystal.kt [hunk 1]
```diff
@@ -5,13 +5,25 @@ import com.fairchild_superconductor.magic_crystal.computer.ComputerEntity
 import net.fabricmc.api.ModInitializer
 import net.fabricmc.fabric.api.`object`.builder.v1.block.FabricBlockSettings
 import net.fabricmc.fabric.api.`object`.builder.v1.block.entity.FabricBlockEntityTypeBuilder
+import net.fabricmc.fabric.api.biome.v1.BiomeModifications
+import net.fabricmc.fabric.api.biome.v1.BiomeSelectors
```
Comment (longfangsong) [inline src/main/kotlin/com/fairchild_superconductor/magic_crystal/MagicCrystal.kt:9]: Your IDE may say this is `Deprecated`, but in fact it is `Experimental`.
I don't think Fabric author use `@Deprecated` correctly.

File: src/main/kotlin/com/fairchild_superconductor/magic_crystal/MagicCrystal.kt [hunk 2]
```diff
@@ -20,34 +32,73 @@ var COMPUTER_ENTITY: BlockEntityType<ComputerEntity>? = null
 object MagicCrystal : ModInitializer {
     const val MOD_ID = "magic_crystal"
     val COMPUTER_BLOCK: ComputerBlock = ComputerBlock(FabricBlockSettings.of(Material.STONE).hardness(4.0f))
+    val TIN_ORE_BLOCK: Block = Block(FabricBlockSettings.of(Material.STONE).hardness(5.0f))
```
Comment (longfangsong) [inline src/main/kotlin/com/fairchild_superconductor/magic_crystal/MagicCrystal.kt:35]: `hardness(5.0f)` might be too hard, try it and suggest one if you want.

File: src/main/kotlin/com/fairchild_superconductor/magic_crystal/MagicCrystal.kt [hunk 3]
```diff
@@ -20,34 +32,78 @@ var COMPUTER_ENTITY: BlockEntityType<ComputerEntity>? = null
 object MagicCrystal : ModInitializer {
     const val MOD_ID = "magic_crystal"
     val COMPUTER_BLOCK: ComputerBlock = ComputerBlock(FabricBlockSettings.of(Material.STONE).hardness(4.0f))
+    val TIN_ORE_BLOCK: Block = Block(FabricBlockSettings.of(Material.STONE).hardness(1.5f))
```
Comment (zhoudian64) [inline src/main/kotlin/com/fairchild_superconductor/magic_crystal/MagicCrystal.kt:35]: Should we define hardness in a more table-driven way? I think that will help Document team.

**Review by zhoudian64 — APPROVED** [review]

File: src/main/kotlin/com/fairchild_superconductor/magic_crystal/MagicCrystal.kt (hunk 3 above)
Comment (longfangsong) [reply src/main/kotlin/com/fairchild_superconductor/magic_crystal/MagicCrystal.kt:35]: We can use regex to get the hardness.

**Pull request merged**